miner/src/pool/scoring.rs

```diff
@@ -108,6 +204,16 @@ impl<P> txpool::Scoring<P> for NonceAndGasPrice where P: ScoredTransaction + txp
 					super::Priority::Regular => 0,
 				};
 				scores[i] = scores[i] << boost;
+				let last_index = txs.len() - 1;
+				for idx in 1..last_index {
```

Comment:
Note the score depends on the next score (`score[idx]`) so we should compute it starting from the end